src/config.ts

```diff
@@ -27,95 +39,116 @@ export class Config {
   notFoundFn: CallableFunction | null;
   cacheEnv: GCache;
   cacheSecretsmanager: GCache;
-
-  constructor(
-    {
-      awsPrefix = "",
-      secretsmanagerPrefix = "",
-      notFoundFn = null,
-    }: {
-      awsPrefix: string,
-      secretsmanagerPrefix: string,
-      notFoundFn: CallableFunction,
-    }
-  ) {
+  cfg: EnvConfig;
+
+  constructor({
+    awsPrefix = "",
+    secretsmanagerPrefix = "",
+    notFoundFn = null,
+  }: {
+    awsPrefix: string;
+    secretsmanagerPrefix: string;
+    notFoundFn: CallableFunction;
+  }) {
     this.awsPrefix = awsPrefix;
     this.secretsmanagerPrefix = secretsmanagerPrefix;
     this.secretsmanagerClient = null;
     this.notFoundFn = notFoundFn;
     this.cacheEnv = new GCache();
     this.cacheSecretsmanager = new GCache();
+    this.cfg = this.getEnvConfig();
+    log("%O", this);
+  }
+
+  getEnvConfig(): EnvConfig {
+    const makeKey = (key: string) => `${this.awsPrefix}${key}`;
+    const keys = {
+      awsAccessKeyId: makeKey("AWS_ACCESS_KEY_ID"),
+      awsSecretAccessKey: makeKey("AWS_SECRET_ACCESS_KEY"),
+      awsRegion: makeKey("AWS_REGION"),
+      awsSessionToken: makeKey("AWS_SESSION_TOKEN"),
+      awsRoleArn: makeKey("AWS_ROLE_ARN"),
+      awsRoleSessionName: makeKey("AWS_ROLE_SESSION_NAME"),
+    };
+    log("keys: %O", keys);
```

Comment:
to confirm, we're not leaking any values here, this is just a list of keys?

Reply:
correct